SimCalorimetry/HGCalSimProducers/src/HGCDigitizerBase.cc

```diff
@@ -70,38 +80,41 @@ void HGCDigitizerBase<DFr>::runSimple(std::unique_ptr<HGCDigitizerBase::DColl>&
     HGCSimHitDataAccumulator::iterator it = simData.find(id);
     HGCCellInfo& cell = (simData.end() == it ? zeroData : it->second);
     addCellMetadata(cell, theGeom, id);
+    double randStep = CLHEP::RandFlat::shoot(0.0, 1.0);
+    long hash_index;
+    if (randStep > 0.5)
+      hash_index = std::abs(CLHEP::RandFlat::shootInt(engine, (NoiseArrayLength_ - 1)) + long(id)) % NoiseArrayLength_;
+    else
+      hash_index = std::abs(CLHEP::RandFlat::shootInt(engine, (NoiseArrayLength_ - 1)) - long(id)) % NoiseArrayLength_;
+    const auto& cellNoiseArray = GaussianNoiseArray_[hash_index];
 
     for (size_t i = 0; i < cell.hit_info[0].size(); i++) {
       double rawCharge(cell.hit_info[0][i]);
-
+      float randNum = cellNoiseArray[i];
       //time of arrival
       toa[i] = cell.hit_info[1][i];
       if (myFEelectronics_->toaMode() == HGCFEElectronics<DFr>::WEIGHTEDBYE && rawCharge > 0)
         toa[i] = cell.hit_info[1][i] / rawCharge;
 
-      //convert total energy in GeV to charge (fC)
-      //double totalEn=rawEn*1e6*keV2fC_;
       float totalCharge = rawCharge;
-
       //add noise (in fC)
       //we assume it's randomly distributed and won't impact ToA measurement
       //also assume that it is related to the charge path only and that noise fluctuation for ToA circuit be handled separately
-      if (noise_fC_[cell.thickness - 1] != 0)
-        totalCharge +=
-            std::max((float)CLHEP::RandGaussQ::shoot(engine, 0.0, cell.size * noise_fC_[cell.thickness - 1]), 0.f);
+      if (noise_fC_[cell.thickness - 1] != 0) {
+        totalCharge += std::max((float)(randNum * noise_fC_[cell.thickness - 1]), 0.f);
```

Comment:
There may be a mistake in this line. The previous Gaussian generation had `cell.size * noise_fC_[cell.thickness - 1]` as the standard deviation parameter. This line just uses `noise_fC_[cell.thickness - 1]`, i.e. the `cell.size` factor is missing.